ios/PurchasesHybridCommon/PurchasesHybridCommonIntegrationTests/CommonFunctionality+async.swift

```diff
@@ -19,6 +32,23 @@ extension CommonFunctionality {
         }
     }
 
+    static func currentOffering(forPlacement placement: String) async throws -> [String: Any]? {
+        let value: PlacementOffering = try await withCheckedThrowingContinuation { continuation in
+            Self.getCurrentOffering(forPlacement: placement) { dictionary, error in
+                let v: PlacementOffering
```

Comment:
I'm not fully understanding this but maybe it's my swift 😅. Why is it needed to map the dictionary back to a `PlacementOffering` before sending it as part of the `Result`?

Reply:
This got me too 😅 An error gets thrown if a `Result` is created like `Result(nil, nil)` which _is_ what we want with an offerings by placement because `nil` is a valid value. So to get _around_ that, we need to map a `nil` to temporary enum to pass into the `Result` and then map that back for the return. It's... a weird Swift thing 😬